### PR TITLE
Create the beginning of a Git/SSH structure for advanced Tower lab

### PR DESCRIPTION
**Pull request by ericzolf**
##### SUMMARY

Create the beginning of a Git/SSH structure for advanced Tower lab.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

- provisioner

##### ADDITIONAL INFORMATION

I didn't yet test it, I wanted first that someone has a look at it and tells me if I did something obviously stupid in the context of the workshops framework. At the end, I decided to amend the advanced_tower.yml playbook, it does hurt less my feelings about good Ansible code... I also added two git_* variables with meaningful defaults.

### CODE REVIEW

File: provisioner/advanced_tower.yml [hunk 1]
```diff
@@ -1,10 +1,71 @@
...
+    package:
+      name:
+      - tree
+      - git
+      state: latest
+
+  - name: make sure git-shell is in /etc/shells
+    lineinfile:
+      dest: /etc/shells
+      line: /usr/bin/shell
```
Comment (cbolz) [inline provisioner/advanced_tower.yml]: this should be /usr/bin/git-shell, no?
@ericzolf

File: provisioner/advanced_tower.yml [hunk 2]
```diff
@@ -1,10 +1,71 @@
...
+  # If the directory exists, we don't want to blow away its
+  # contents, because it's probably got valid repo data in it!
+
+  - name: Check Git project directory for existence
+    stat:
+      path: /home/{{ git_user }}/{{ git_project }}.git
+    register: project_dir
+
+  # If the directory exists, we don't want to blow away its
+  # contents, because it's probably got valid repo data in it!
```
Comment (cbolz) [inline provisioner/advanced_tower.yml]: this is the same comment as in line 43 and 44 - duplicate?

**Review by cbolz — CHANGES_REQUESTED** [review]

File: provisioner/advanced_tower.yml (hunk 1 above)
Comment (ericzolf) [reply provisioner/advanced_tower.yml]: Fixed

File: provisioner/advanced_tower.yml [hunk 3]
```diff
@@ -1,10 +1,71 @@
...
+  - name: add the newly created SSH key to authorized keys
+    authorized_key:
+      user: "{{ git_user }}"
+      state: present
+      key: "{{ git_authorized_key['content'] | b64decode }}"
+
+  # If the directory exists, we don't want to blow away its
+  # contents, because it's probably got valid repo data in it!
+
+  - name: Check Git project directory for existence
```
Comment (cbolz) [inline provisioner/advanced_tower.yml:54]: it would probably be good to have a sub directory, in case the student wants to create a second repo

Comment (ericzolf) [reply provisioner/advanced_tower.yml:54]: It's probably not necessary but then I'll fix also the duplicate comments...

File: provisioner/advanced_tower.yml [hunk 4]
```diff
@@ -1,10 +1,81 @@
...
+  - name: Ensure {{ git_project }}.git owned by git
+    file:
+      path: "{{ projects_dir }}/{{ git_project }}.git"
+      owner: "{{ git_user }}"
+      group: "{{ git_user }}"
+      recurse: yes
+      mode: ug+rwX
+      state: directory
+
+  - name: add user {{ username }} to {{ git_user }} group
```
Comment (cbolz) [inline provisioner/advanced_tower.yml]: username is a hostvar and should not be used in the task, better use something like:
- name: add user to {{ git_user }} group
or 
- name: add student to {{ git_user }} group

**Review by cbolz — CHANGES_REQUESTED** [review]
find another small cosmetic glitch

File: provisioner/advanced_tower.yml (hunk 4 above)
Comment (ericzolf) [reply provisioner/advanced_tower.yml]: Modified...

**Review by cbolz — APPROVED** [review]

**Pull request merged**